### PR TITLE
Update sidekiq-prometheus-exporter and migrate to puma for serving requests

### PR DESCRIPTION
**Pull request by ThisIsMissEm**
This fixes the issue with the spurious 5xx errors, but also:
- Aligns dependencies on sidekiq & redis with Mastodon
- Switches from rackup to puma (we had to run an outdated rack version to use rackup, but we still run as a single process)
- Hardens the systemd unit file, and has it use rbenv from the mastodon user.
- Adds basic apache style request logging, so we can see the service is working

### CODE REVIEW

File: etc/systemd/system/sidekiq-prometheus-exporter.service [hunk 1]
```diff
@@ -3,10 +3,48 @@ Description=Sidekiq Prometheus Exporter
 After=network.target
 
 [Service]
+User=mastodon
 WorkingDirectory=/var/lib/sidekiq-prometheus-exporter
-ExecStart=bundle exec rackup -p31337 -o0.0.0.0
+Environment="PUMA_ENV=production"
+Environment="BIND=0.0.0.0"
+Environment="PORT=31337"
```
Comment (ThisIsMissEm) [inline etc/systemd/system/sidekiq-prometheus-exporter.service:10]: @Preskton we should probably make these set via an override file. I'm not sure why we need this listening on 0.0.0.0 though?

File: etc/systemd/system/sidekiq-prometheus-exporter.service [hunk 2]
```diff
@@ -3,10 +3,48 @@ Description=Sidekiq Prometheus Exporter
...
+ExecStart=/home/mastodon/.rbenv/shims/bundle exec puma
```
Comment (ThisIsMissEm) [inline etc/systemd/system/sidekiq-prometheus-exporter.service:11]: This ensures we're using the same ruby version as what we're using with Mastodon.

File: config.ru [hunk 3]
```diff
@@ -8,5 +8,6 @@ Sidekiq.configure_client do |config|
   config.redis = {url: ENV['REDIS_URL']}
```
Comment (ThisIsMissEm) [inline config.ru:8]: We may want to cascade this as `SIDEKIQ_REDIS_URL` then to `REDIS_URL`, for a scenario where we have sidekiq running on a separate redis to cache/timelines.

File: etc/systemd/system/sidekiq-prometheus-exporter.service (hunk 1 above)
Comment (Preskton) [reply etc/systemd/system/sidekiq-prometheus-exporter.service:10]: Yup, on the host we are binding to localhost.

**Review by Preskton — APPROVED** [review]

**Pull request merged**